WORKFLOW.md

```diff
@@ -0,0 +1,137 @@
+X-Road Tests development
+# Workflow Policy
+
+v1.0
+
+## 1	General
+
+1.1	This document establishes workflow policy for joint development of X-Road software. Workflow Policy strives to serve as a handbook for all practical aspects of development.
+
+1.2	Workflow is understood as systematic arrangement of work, a complex of processes, practices, roles and responsibilities, communication patterns, and artefacts.
+
+1.3	Goals of the workflow policy are to
+- establish productive and secure collaborative open source working environment
+- assure production of high quality software
+- avoid duplication of effort, facilitate re-use of software
+- transparency and openness
+- broader community of developers
+- use of software development best practice
+- clear communication among Partners as well as other stakeholders in the X-Road development process
+- innovation.
+
+1.4 Adherence to this policy are made legally binding to Vendors by inclusion of appropriate stipulations in contracts between Partners and Vendors. Vendor in context of this policy is a firm or other organisation performing development work by assigment of a Partner.
+
+1.5 Partners and Developers implement this policy in good faith and in the context of sustainable, good software development practice.
+
+1.6 Projects may have their own workflow arrangements as far as these do not contradict this policy.
+
+## 2 Related documents
+
+2.1 Workflow policy is related to other X-Road tests development policy and regulations:
+- workflow policy implements Harmonized Xroad tests document [Harmonized X-Road test environment](https://github.com/ria-ee/blob/master/HARMONIZED_TEST_ENVIRONMENT.md)
+
+## 3 Development model
+
+3.1	X-Road tests uses [Github](https://github.com/) and [Git](https://git-scm.com/) based version management.
+
+## 4	Repositories
+
+4.1	The following code and documentation repositories are used:
+- Master Test Repository - short name: X-Road-tests; hosted by: GitHub; managed by: RIA; purpose: release of X-Road automated tests; access: Head Architect has write access; read access: ALL.
+	- https://github.com/ria-ee/X-Road-tests
+
+- Finland and Estonia develop in fork of ria-ee/X-Road-tests
+	- Finland fork
+		- https://github.com/vrk-kpa/X-Road-tests
+	- Estonia fork
+		- https://github.com/asaquality/X-Road-tests
+
+4.2	Partners can establish their own, additional repositories, for backup, software distribution or other purposes.
+
+## 5	Branching pattern
+
+5.1	Branching pattern follows the [Gitflow model](https://www.atlassian.com/git/tutorials/comparing-workflows/gitflow-workflow). Two perpetual branches – `master` , `develop` and own `testfeature` branches.
+
+- `master` branch is used to release X-Road tests
+- `develop` branch is used to test development
+- `feature` branches are used to work on test features
+
+Main repository:
+- `master` branch https://github.com/ria-ee/X-Road-tests
+	- Tagged releases of Xroad-tests
```

Comment:
It should be specified that every release in master of X-Road-tests corresponds to a specific release of X-Road software. Currently it is not as clear. We could also agree on some specific way to indicate the relation with X-Road code. It may be a link to X-Road tag (for master) or commit (for develop) in root README.md of X-Road-tests etc.

Reply:
yes. How i should describe this? Any proposals?
It should be specified that every release in master of X-Road-tests corresponds to a specific release of X-Road softwar

Reply:
- master branch https://github.com/ria-ee/X-Road-tests/tree/master
  * releases of X-Road-tests that correspond to a specific release of X-Road source code. Several versions of X-Road-tests may point to the same X-Road source code version if necessary.
- develop branch https://github.com/ria-ee/X-Road-tests/tree/develop
  * Integration branch between Finland and Estonia (and other possible partners?)
  * Latest integrated test development source code
  * Make pull request to master for tagging release or milestone

README.md file in the root of X-Road-tests must always point to the specific tag or commit of the X-Road source code for which these tests were designated for.

Reply:
I agree that it has to be defined in README.md file for which X-Road source code version these tests were designated for. Must be clearly marked down (in README.md) for both MASTER and DEVELOP branches so that no confusion appears.

For clarity, wording for the last sentence could be something like this:
_README.md file in the root of X-Road-tests must indicate the X-Road source code version for both MASTER and DEVELOP branches so that there is no confusion for which version these tests were designated for. Indication is done by pointing to the specific tag (MASTER) or commit (DEVELOP) of the X-Road source code._